.cicdtemplate/.github/workflows/deploy_staging_and_production.yml

```diff
@@ -1,29 +1,32 @@
-name: Deploy Staging
+name: Deploy staging and production
 
 on:
   # Trigger the workflow on push action in develop branch
   # So it will trigger when the PR of the feature branch was merged.
   push:
     branches:
       - develop
+
 jobs:
-  app_deploy:
-    name: Deploy and upload to Firebase App Distribution
+  deploy_staging_and_production:
+    name: Deploy staging and production
     runs-on: ubuntu-latest
+    timeout-minutes: 30
```

Comment:
Hmmm, when applying this template to projects, developers would need to update this? 🤔  I am still thinking if we really need the timeout. cc @lydiasama

Reply:
Hmmmm, so if this job takes more than 30 minutes it will get canceled? 🤔 

That could lead to a problem once a project starts growing

Reply:
I think it's ok the timeout value can be updated later; imagine if the job is failed and you CAN read the error message (right?) then as the maintainer, you should be able to update it. In the beginning, this will guard us for a very long, unwanted pipeline that costs us unwanted expense 🤔

Reply:
I think better to put a comment here to remind about the update 🤔

Reply:
😂  there will be an error message, simply read and retry....I don't think a comment is helpful here...like...if you read this first before the error message then it's okay, other than that, I'd recommend a developer should read the message, understand it and the configuration before making changes...we don't always offer comment everywhere to hand-holding everything...